### PR TITLE
add initial 'data' to plugin context

### PR DESCRIPTION
**Pull request by arcadianCdr**
adds 'data' in plugin context based on pv's object data for first plugin run (was previously empty).

### CODE REVIEW

File: src/plugins/events/actor.ts [hunk 1]
```diff
@@ -23,7 +24,21 @@ export async function onActorCreate(
 
   const createdImages = [] as Image[];
 
-  const pluginResult = await runPluginsSerial(config, event, {
+  const labels = (await Actor.getLabels(actor)).map((l) => l.name);
```
Comment (boi123212321) [inline src/plugins/events/actor.ts:27]: Just return the labels, not just their names

File: src/plugins/events/actor.ts [hunk 2]
```diff
@@ -23,7 +24,21 @@ export async function onActorCreate(
...
+    name: actor.name,
+    description: actor.description ? actor.description : undefined,
+    bornOn: actor.bornOn ? actor.bornOn : undefined,
+    addedOn: actor.addedOn ? actor.addedOn : undefined,
+    rating: actor.rating ? actor.rating : undefined,
+    favorite: actor.favorite,
+    bookmark: actor.bookmark ? actor.bookmark : undefined,
+    nationality: actor.nationality ? actor.nationality : undefined,
+    aliases: actor.aliases?.length ? actor.aliases : undefined,
+    labels: labels?.length ? labels : undefined,
```
Comment (boi123212321) [inline src/plugins/events/actor.ts:38]: To make it more consistent, just use labels, that way it's always an array

File: src/plugins/events/actor.ts [hunk 3]
```diff
@@ -23,7 +24,21 @@ export async function onActorCreate(
...
+  const initialData: Dictionary<unknown> = {
+    name: actor.name,
+    description: actor.description ? actor.description : undefined,
+    bornOn: actor.bornOn ? actor.bornOn : undefined,
+    addedOn: actor.addedOn ? actor.addedOn : undefined,
+    rating: actor.rating ? actor.rating : undefined,
+    favorite: actor.favorite,
+    bookmark: actor.bookmark ? actor.bookmark : undefined,
+    nationality: actor.nationality ? actor.nationality : undefined,
+    aliases: actor.aliases?.length ? actor.aliases : undefined,
```
Comment (boi123212321) [inline src/plugins/events/actor.ts:37]: Same as labels

File: src/plugins/events/actor.ts [hunk 4]
```diff
@@ -23,7 +24,21 @@ export async function onActorCreate(
 
   const createdImages = [] as Image[];
 
-  const pluginResult = await runPluginsSerial(config, event, {
+  const labels = (await Actor.getLabels(actor)).map((l) => l.name);
+  const initialData: Dictionary<unknown> = {
+    name: actor.name,
+    description: actor.description ? actor.description : undefined,
```
Comment (boi123212321) [inline src/plugins/events/actor.ts:30]: I don't really understand the switch to undefined here, why not just use description?
Same for all the other properties

File: src/plugins/events/scene.ts [hunk 5]
```diff
@@ -61,7 +62,28 @@ export async function onSceneCreate(
...
+    views: views || undefined,
+    watches: watches.length ? watches : undefined,
+    rating: scene.rating ? scene.rating : undefined,
+    favorite: scene.favorite,
+    bookmark: scene.bookmark ? scene.bookmark : undefined,
+    actors: actors?.length ? actors : undefined,
+    labels: labels?.length ? labels : undefined,
+    studio: scene.studio ? (await Studio.getById(scene.studio))?.name : undefined,
+    // If more than one movie, uses the first one
+    movie: (await Movie.getByScene(scene._id))?.[0]?.name,
```
Comment (boi123212321) [inline src/plugins/events/scene.ts:83]: Just return all movies, let the plugin use the first one if it wants to

File: src/plugins/index.ts [hunk 6]
```diff
@@ -38,11 +39,12 @@ export async function runPluginsSerial(
     logger.info(`Running plugin ${pluginName}:`);
     try {
       const pluginResult = await runPlugin(config, pluginName, {
-        data: <typeof result>JSON.parse(JSON.stringify(result)),
+        data: initialData ?? <typeof result>JSON.parse(JSON.stringify(result)),
         event,
         ...inject,
         pluginArgs,
       });
+      initialData = undefined;
```
Comment (boi123212321) [inline src/plugins/index.ts:47]: Why?

File: src/plugins/index.ts [hunk 7]
```diff
@@ -38,11 +39,12 @@ export async function runPluginsSerial(
     logger.info(`Running plugin ${pluginName}:`);
     try {
       const pluginResult = await runPlugin(config, pluginName, {
-        data: <typeof result>JSON.parse(JSON.stringify(result)),
+        data: initialData ?? <typeof result>JSON.parse(JSON.stringify(result)),
```
Comment (boi123212321) [inline src/plugins/index.ts:42]: This way only the first plugin has access to the data, maybe it should be provided under a different property, that way any plugin can differentiate between piped (new) data, and the initial data

**Comment by arcadianCdr** [comment]
Modified according to our discussions to:
- Server functions to get the extra initial data: lazy loaded on first use (an result reused for subsequent calls as they are immutable).
- Removed compatibility with the plugin's output types to allow the full details of the object properties and the extra server functions => removed the initialData structure as they would have been a duplicate with the full initial object that is already passed to the context.
- As a help to plugin developers, I added a new getMergedData() method on the plugin side. It converts the initial data to the output type and merge with data (also in output type).

**Review by leadwolf — APPROVED** [review]

**Pull request merged**